normalize/diff.py

```diff
@@ -660,6 +664,20 @@ def compare_collection_iter(propval_a, propval_b, fs_a=None, fs_b=None,
             rev_key[(pk, seen[pk])] = k
             seen[pk] += 1
 
+    if options.recurse:
+        # we can be sure that both records have these keys
+        shared_keys = set(rev_key.values())
```

Comment:
That would be something like: `set(rev_keys["a"].values()).intersection(rev_keys["b"].values())`